### ISSUE TITLE
Seg fault reading NASA GOLD data

### ISSUE THREAD

**Issue by bharding512** [Unidata/netcdf4-python#915]
New data from the NASA GOLD mission just came out. netcdf4-python seg faults when reading these data:

Example file (40 MB)
https://spdf.sci.gsfc.nasa.gov/pub/data/gold/level1c/2019/001/GOLD_L1C_CHA_DAY_2019_001_07_52_v01_r01_c01.nc

```
import netCDF4
fn = '/path/to/GOLD_L1C_CHA_DAY_2019_001_21_40_v01_r01_c01.nc'
d = netCDF4.Dataset(fn)
print(d)
 ```

The problem appears to be with the variable attribute VAR_NOTES which is set to NIL. It is possible to load the numerical data, but it seems that reading the attribute VAR_NOTES is what causes the seg fault:
```
import netCDF4
fn = '/path/to/GOLD_L1C_CHA_DAY_2019_001_21_40_v01_r01_c01.nc'
d = netCDF4.Dataset(fn)
x = d.variables['RADIANCE'][...] # this works fine
print(d.variables['RADIANCE'].VAR_NOTES) # this seg faults
```

Some other notes:

- Python 2.7 and 3.7 tested
- netCDF4 v1.4.0 and v1.5.0.1 tested
- Loading with xarray also causes segfault
- `ncdump` correctly prints VAR_NOTES as NIL

Thanks.

**Comment by jswhit** [comment]
Confirmed.  Tentative fix in issue #916 (branch `issue915`).

**Comment by jswhit** [comment]
Pull request #916 merged, so this is now fixed in master.  Closing now, feel free to re-open if issue persists.  I will release 1.5.1 next week.
